yarn-project/simulator/src/public/public_processor/public_processor.ts

```diff
@@ -240,16 +243,6 @@ export class PublicProcessor implements Traceable {
           continue;
         }
 
-        if (!tx.hasPublicCalls()) {
```

Comment:
I moved this to the privateOnly method. Since we already have one, why not just encapsulate everything into either the privateOnly method or the public method.